packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/useGridRowGroupingPreProcessors.ts

```diff
@@ -17,7 +17,7 @@ import {
   createRowTree,
   updateRowTree,
   RowTreeBuilderGroupingCriterion,
-  getVisibleRowsLookup,
+  getHiddenRowsLookup,
```

Comment:
> However, making that change lost nearly all of the performance improvements observed in the first step.

I don't understand what caused the loss of the performance improvements.
First, you tried using `Set` to keep all the visible rows.
Then, you reversed it to only keep the hidden rows. Did this increase the number of records in the `Set` drastically?
Or is it something else that impacted performance?

Reply:
Also didn't make sense to me, but I haven't had enough time to investigate further.